### DISCUSSION TITLE
RPC Method Not Found

### DISCUSSION THREAD

**Discussion by 0xatillaa** [WalletConnect/walletconnect-monorepo#3401] [Developer Support]
**Describe the bug**
Trying to sign messages doesn't work, normal transactions work. I am trying to create a WalletConnect v2.0 connector for web3-react/v6 and normal transactions (approve, swap etc) work normally but cannot make signatures. Wondering whether I need to get sign-client involved or something else to make this work. I am trying to do this on Moonbeam.

**SDK Version (if relevant)**
 "@walletconnect/ethereum-provider": "^2.7.4",
  @web3modal/standalone": "^2.4.1",

**To Reproduce**
1. Click anything that requires signing an EIP712 message
2. get Method not found output

**Expected behavior**
Allow me to sign

**Screenshots**

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/48114560/77a80adf-df6e-4e50-8276-8e03f8794b12)

**Desktop (please complete the following information):**
 - OS: [Linux]
 - Browser [Chrome]
 - Version [110]

Below is the code:

```

**Comment by ganchoradkov** [comment]
Hey @atillabirer could you share your request structure? Which methods are you using in your requests that produce `Method not found` exception?

**Comment by 0xatillaa** [comment]
> Hey @atillabirer could you share your request structure? Which methods are you using in your requests that produce `Method not found` exception?

I am using eth_signTypedData_v4 , which works with Metamask. On my connector in the rpc list I have tried to add "eth_signTypedData_v4" but then it gives "method not found" on multiple wallets. eth_signTypedData doesnt work very well with eip712 structures with arrays.

**Comment by ganchoradkov** [comment]
`eth_signTypedData_v4` is not likely to be supported by the majority of wallets. 
Is there any reason you're not using `personal_sign`? It should be supported by pretty much every wallet

**Comment by 0xatillaa** [comment]
> `eth_signTypedData_v4` is not likely to be supported by the majority of wallets. Is there any reason you're not using `personal_sign`? It should be supported by pretty much every wallet

It's the default RPC method used by ethers.Signer, y metamask silently fails v3, and also it's necessary in order to work with EIP712 signatures that have array and struct values, lack of this breaks Permit2 which is a huge security feature now with approvals that are valid for only 1 transaction (by signature). Uniswap uses permit2 for almost everything now, please implement this or I can contribute to the implementation if you want, but us and many walletconnect users need this working.

**Comment by ganchoradkov** [comment]
v2 doesn't have specific handling for different methods including `eth_signTypedData_v4` and can transmit any transaction (up to websocket limits) between the dapp & wallet.
It's up to the wallets to support it so I'm not sure how we can help you in this case.

**Comment by jeremy-kunzhou** [comment]
Working on merge from WC1 to WC2 and found this problem via using permit2 related smart contract methods.

Environment: WC2  "@web3modal/react-native": "^1.0.0-alpha.5",
wallet: keyring pro

get this error:
error: {"code": -32601, "message": "the method eth_signTypedData_v4 does not exist/is not available"}
I tried to add "eth_signTypedData_v4" to the namespaces in session params, but it is not helpful.
```
export const sessionParams = (supportChains: number[]) => ({
  namespaces: {
    eip155: {
      methods: [
        "eth_sendTransaction",
        "eth_signTransaction",
        "eth_sign",
        "personal_sign",
        "eth_signTypedData",
        "eth_signTypedData_v4",
      ],
      chains: supportChains.map((item) => `eip155:${item}`),
      events: ["chainChanged", "accountsChanged"],
      rpcMap: {},
    },
  },
})
```

it works fine using WC1 with same wallet.
Have not try with other wallet but based on current experiment, not too much handy mobile wallet support WC2 well in react native development process on android.

**Comment by TimDaub** [comment]
By upgrading to wagmi 1.0.0, I'm starting to have the same issue for a signing script that was previously working with wagmi 0.12.x

Here's my updated code base: https://github.com/attestate/kiwistand/tree/upgrade-wagmi-to-1.0

And here's that part of the code that fails for the same reasons as above state: https://github.com/attestate/kiwistand/blob/e8f50c899705f10bc1b3325e3979b2d05a4a9411/src/web/src/Vote.jsx#L23

**Comment by cosiato** [comment]
Same issue here.

I was able to sign EIP712 messages with wallets like Trustwallet, Zengo or Ledger with WalletConnect v1. 

Now, it instantly fails when going through WalletConnect v2 with the aforementioned wallets with the following error : 
`Missing or invalid. request() method: eth_signTypedData_v4”`. 

[WalletConnect docs](https://docs.walletconnect.com/2.0/advanced/rpc-reference/ethereum-rpc#eth_signtypeddata) shows that RPC method used for Typed Data is **eth_signTypedData**, is it the reason why it fails ? 


My config: 
wagmi 1.0.9 -> their useSignTypedData hook uses **eth_signTypedData_v4** method.

**Comment by glitch-txs** [comment]
I’m closing this issue because it has been inactive for a few months.

Please reopen if you still encounter this issue with the latest version :)

**Comment by slavik0329** [comment]
This issue still exists. Cannot use eth_signTypedData_v4 with walletconnect v2

**Comment by glitch-txs** [comment]
Hi @slavik0329 could you provide more information in your use case? So far we've tested eth_signTypedData_v4 and haven't found this issue:
- What are you version and packages?
- Which wallet are you testing this with?
- Could you share a minimal reproducible example of this issue + steps to reproduce?

This method should work as long as it's supported by the wallet. Thanks!
